src/objects/string.rs

```diff
@@ -583,4 +610,47 @@ mod test {
         let v = py_bytes.extract::<Vec<u8>>(py).unwrap();
         assert_eq!(b"Hello", &v[..]);
     }
+
+    #[test]
+    fn test_extract_umlaut() {
+        let gil = Python::acquire_gil();
+        let py = gil.python();
+        let py_string = py.eval("u'x=\\u00e4'", None, None).unwrap();
+        let data = py_string.cast_as::<PyString>(py).unwrap().data(py);
+        #[cfg(feature = "python3-sys")]
+        {
+            if let PyStringData::Latin1(s) = data {
```

Comment:
Is it worth adding tests for the other `PyStringData` variants?  I assume non-Latin-1 text (e.g. something in Greek) would be Utf16 and non-BMP text (e.g. something with emojis) would be Utf32.